### PR TITLE
Submission Service to CFX

### PR DESCRIPTION
**Pull request by KFloodBCGov**
# Description

This PR includes the following proposed change(s):

Switched the submission service to CFX, implementedthe SOAP call
Change Mock Impl to Demo Impl and update worker application to use it.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Local build and unit tests.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/libs/efiling-submission-client/src/main/java/ca/bc/gov/open/jag/efilingsubmissionclient/config/AutoConfiguration.java [hunk 1]
```diff
@@ -1,40 +1,44 @@
...
+            URL url = new URL(csoSubmissionProperties.getFilingSubmissionSoapUri());
+            filingFacade = new FilingFacade(url, serviceName);
+        } catch(MalformedURLException e) {
 
-        CSOSubmissionServiceImpl eFilingSubmissionServiceImpl = new CSOSubmissionServiceImpl();
-        eFilingSubmissionServiceImpl.setDefaultUri(csoSubmissionProperties.getFilingSubmissionSoapUri());
-        eFilingSubmissionServiceImpl.setMarshaller(jaxb2Marshaller);
-        eFilingSubmissionServiceImpl.setUnmarshaller(jaxb2Marshaller);
-        return eFilingSubmissionServiceImpl;
+            LOGGER.error("Malformed URL exception :" + e.getMessage());
```
Comment (TayGov) [inline src/backend/libs/efiling-submission-client/src/main/java/ca/bc/gov/open/jag/efilingsubmissionclient/config/AutoConfiguration.java:40]: For errors you can just do ("message", e)

File: src/backend/libs/efiling-submission-client/src/main/java/ca/bc/gov/open/jag/efilingsubmissionclient/config/AutoConfiguration.java [hunk 2]
```diff
@@ -1,40 +1,44 @@
...
 
     @Bean
     @ConditionalOnMissingBean(value = {EfilingSubmissionService.class})
-    public EfilingSubmissionService eFilingSubmissionClient(@Qualifier("CSOSubmissionMarshaller") Jaxb2Marshaller jaxb2Marshaller) {
+    public FilingFacade eFilingSubmissionService() {
+
+        FilingFacade filingFacade = null;
+        try {
+
+            QName serviceName = new QName("http://filing.csows.ag.gov.bc.ca/", "FilingFacade");
```
Comment (TayGov) [inline src/backend/libs/efiling-submission-client/src/main/java/ca/bc/gov/open/jag/efilingsubmissionclient/config/AutoConfiguration.java:35]: Is this going to be a property?

**Comment by akroon3r** [comment]
Good job on the diff %

File: src/backend/libs/efiling-submission-client/src/test/java/ca/bc/gov/open/jag/efilingsubmissionclient/CsoSubmissionServiceImplTest.java [hunk 3]
```diff
@@ -20,7 +17,9 @@
 @DisplayName("CSO Submission Service Suite")
 public class CsoSubmissionServiceImplTest {
 
-    @InjectMocks
+    private static final String CASE_1 = "CASE_1";
+    private static final String CASE_2 = "CASE 2";
```
Comment (akroon3r) [inline src/backend/libs/efiling-submission-client/src/test/java/ca/bc/gov/open/jag/efilingsubmissionclient/CsoSubmissionServiceImplTest.java:21]: missing an underscore here I think

Comment (akroon3r) [reply src/backend/libs/efiling-submission-client/src/test/java/ca/bc/gov/open/jag/efilingsubmissionclient/CsoSubmissionServiceImplTest.java:21]: or vice versa

Comment (akroon3r) [reply src/backend/libs/efiling-submission-client/src/test/java/ca/bc/gov/open/jag/efilingsubmissionclient/CsoSubmissionServiceImplTest.java:21]: no biggie

**Pull request merged**